java/statestore-commit/src/test/java/sleeper/commit/StateStoreCommitterTest.java

```diff
@@ -186,6 +188,58 @@ void shouldFailForRetryWhenStateStoreThrowsUnexpectedStateStoreException() throw
         }
     }
 
+    @Nested
+    @DisplayName("Assign job ID to files")
+    class AssignJobIdToFiles {
+        @Test
+        void shouldApplyCompactionJobIdAssignmentCommitRequest() throws Exception {
+            // Given
+            StateStore stateStore = createTable("test-table");
+            FileReference inputFile = fileFactory.rootFile("input.parquet", 123L);
+            stateStore.addFile(inputFile);
+
+            // When
+            CompactionJobIdAssignmentCommitRequest request = new CompactionJobIdAssignmentCommitRequest(List.of(
+                    assignJobOnPartitionToFiles("job1", "root", List.of("input.parquet"))), "test-table");
+            committer().apply(StateStoreCommitRequest.forCompactionJobIdAssignment(request));
+
+            // Then
+            assertThat(stateStore.getFileReferences()).containsExactly(
+                    withJobId("job1", fileFactory.rootFile("input.parquet", 123L)));
+        }
+
+        @Test
+        void shouldIgnoreJobIdAssignmentIfFileReferenceIsAlreadyAssignedToJob() throws Exception {
+            // Given
+            StateStore stateStore = createTable("test-table");
+            FileReference inputFile = withJobId("job1", fileFactory.rootFile("input.parquet", 123L));
+            stateStore.addFile(inputFile);
```

Comment:
It's a bit unclear whether we can expect the state store to respect the job ID assigment when adding a new file here. Please can we tell the state store to assign the job explicitly?